torch/csrc/jit/python/init.cpp

```diff
@@ -122,6 +125,24 @@ using ::c10::FunctionSchema;
 using caffe2::serialize::PyTorchStreamReader;
 using caffe2::serialize::PyTorchStreamWriter;
 
+class PythonSymbolicIntNode: public c10::SymbolicIntNode {
+public:
+  PythonSymbolicIntNode(py::object pyobj): 
+    c10::SymbolicIntNode() {
+      pyobj_ = std::make_shared<c10::SafePyObject>(pyobj.release().ptr(), getPyInterpreter());
+    };
+
+  virtual SymbolicIntNode* add(SymbolicIntNode* other) override {
+    auto pother = dynamic_cast<PythonSymbolicIntNode*>(other);
+    py::gil_scoped_acquire acquire;
+    auto r = getPyObj().attr("__add__")(pother->getPyObj());
+    return new PythonSymbolicIntNode(r);
+  }
+
+  py::handle getPyObj() { return py::handle(pyobj_.get()->ptr(getPyInterpreter())); }
+  std::shared_ptr<c10::SafePyObject> pyobj_ = nullptr;
```

Comment:
I don't think this needs to be a `shared_ptr`, is `PythonSymbolicIntNode` copyable?